src/styles/utils/cycle.scss

```diff
@@ -0,0 +1,24 @@
+// add day number
+@for $dayNumber from 1 through 31 {
+  .calendar__day:nth-child(#{$dayNumber})::before {
+    content: '#{$dayNumber}';
+  }
+}
+
+// delete over days
+@for $delDay from 28 to 31 {
+  .calendar--month-length-#{$delDay} {
+    @for $del from ($delDay + 1) through 31 {
+      & :nth-child(n + #{$del}) {
+        display: none;
+      }
+    }
+  }
+}
+
+// position first day
+@each $name, $value in $daysOfWeek {
+  .calendar--start-day-#{$name} .calendar__day:first-child {
+    margin-left: ($value * ($daySize + $gapValue)) - ($daySize + $gapValue);
+  }
+}
```

Comment:
No, no, no.
Never put cycles in a separate file!
They should be placed with their selectors together